python/pyarrow/array.pxi

```diff
@@ -1197,7 +1197,13 @@ cdef class ListArray(Array):
     def values(self):
         return self.flatten()
 
-    # TODO(wesm): Add offsets property
+    @property
+    def offsets(self):
+        """
+        Return the offsets as an int32 array.
+        """
+        return Array.from_buffers(
+            int32(), len(self) + 1, [None, self.buffers()[1]])
```

Comment:
Should this rather be implemented in C++ (there is currently no method on ListArray in C++ that directly returns the offsets as an arrow::Array)

Reply:
Right, that would be better IMO.

Reply:
Easy to do, picking a name for such a method maybe is the hardest part.

We have

```
ListArray::values -> shared_ptr<Array>
ListArray::value_offset -> int32_t
ListArray::value_offsets -> shared_ptr<Buffer>
ListArray::raw_value_offsets -> const int32_t*
```

Thoughts about what to call it?

Reply:
`ListArray::offsets_array`?